content/pl/dev-prod-parity.md

```diff
@@ -0,0 +1,78 @@
+## X. Jednolitość środowisk
+### Utrzymuj środowisko developerskie, stagingowe, i produkcyjne tak podobne jak tylko możliwe
+
+Historically, there have been substantial gaps between development (a developer making live edits to a local [deploy](./codebase) of the app) and production (a running deploy of the app accessed by end users).  These gaps manifest in three areas:
+
+Z doświadczenia wiadomo, że od zawsze istniały różnice pomiędzy środowiskiem developerskim (developer pracujący nad swoją lokalną wersją [kodu](./codebase) aplikacji) a produkcyjnym (działająca aplikacja dostępna dla użytkowników. Ze względu na charakter, możemy wymienić trzy rodzaje różnic:
+
+* **Różnica czasowa:** Developer może pracować nad kodem przez dni, tygodnie, miesiące zanim ostatecznie pojawi się on w wersji produkcyjnej.
+* **Różnica odpowiedzialności**: Developer tworzy kod aplikacji, natomiast kto inny wdraża go do na produkcję.
+* **Różnica narzędzi**: Developer może używać narzędzi takich jak Nginx, SQLite i systemu OS X, natomiast wersja produkcyjna będzie się opierać na Apache, MySQL i systemie Linux.
+
+**Aplikacja 12factor jest zaprojektowana by można ją było [nieustannie wdrażać na produkcję](http://www.avc.com/a_vc/2011/02/continuous-deployment.html) minimalizując różnice pomiędzy środowiskami.**  Mając na uwadze powyższe różnice, można sobie z nimi radzić:
+
+* Zmniejsz czas deploymentu: czas wdrożenia kodu napisanego przez developera powinien być mierzony w godzinach, a nawet w minutach.
+* Przenieś odpowiedzialność: developer piszący kod powinien być zaangażowany we wdrożenia aplikacji na produkcję.
+* Stosuj ten sam zestaw narzędzi : utrzymuj wszystkie środowiska w których działa aplikacja tak podobne jak to możliwe.
+
+Podsumowując w formie tabeli:
+
+<table>
+  <tr>
+    <th></th>
+    <th>Tradycyjna aplikacja</th>
+    <th>Aplikacja 12factor</th>
+  </tr>
+  <tr>
+    <th>Czas pomiędzy wdrożeniami</th>
+    <td>Tygodnie</td>
+    <td>Godziny</td>
+  </tr>
+  <tr>
+    <th>Tworzenie kodu vs wdrażanie kodu</th>
+    <td>Różne osoby</td>
+    <td>Te same osoby</td>
+  </tr>
+  <tr>
+    <th>Środowisko developerskie vs produkcyjne</th>
+    <td>Mocno różniące się</td>
+    <td>Jak najbardziej zbliżone</td>
+  </tr>
+</table>
+
+[Usługi wspierające](./backing-services), takie jak baza danych aplikacji, system kolejkowania czy cache są polem gdzie zachowanie podobieństw między wdrożeniami jest ważne. Wiele języków oferuje biblioteki, które upraszczają korzystanie z usług wspierających w tym *adaptery* do usług różnego typu. Kilka przykładów w tabeli poniżej:
```

Comment:
1. Przecinek raczej niepotrzebny po `backing-services`

Całe to zdanie źle brzmi. Proponuję coś w stylu `Zachowanie podobieństw między wdrożeniami jest ważne w przypadku usług wspierających takich jak baza danych aplikacji, system kolejkowania czy też cache`